.gitlab-ci.yml

```diff
@@ -120,6 +120,17 @@ test:
   script:
     - cargo test --verbose --all-features --no-fail-fast --workspace
 
+
+docs:
+    stage:                           workspace
+    <<:                              *docker-env
+    script:
+        - output=$( cargo doc --workspace --no-deps 2>&1 )
+        - 'has_warnings=$(egrep -c "^warning: [1-9]+[0-9]* warnings emitted" <<< "$output" || true)'
+        - if [ "$has_warnings" -eq 0 ]; then exit 0; fi
+        - echo "$output"
+        - exit 1
```

Comment:
not needed

Reply:
But won't it take the `$?` from the `echo` then as the last exit code, which would be `0` and thus the job would not show as failed?

Reply:
Did you test if it exits with `$?` I'm not entirely sure
- it shouldn't exit with 1 regardless of what happened, there might be other failures
- it could exit with the other code

Reply:
So I tested it by now (the commit and it's revert commit are in this PR). My understanding ist that we need the `exit 1` to mark the job as a failure and if we leave it out it gets marked as a success. I tested it in the mentioned commits by removing `exit 1` and deliberately introducing an error, the job was still marked as success. Yeah so tl;dr is that I think we need it.